transform/instance.go

```diff
@@ -367,6 +367,75 @@ func (ot *objectTransformer) transform(in interface{}, modifier pathModifier) (i
 
 }
 
+// transform retrieves the value for this object by building the value for the base object and then adding in any
+// transforms for all defined child fields. If a transform is provided it transforms the children relative to the
+// passed in node
+func (ot *objectTransformer) objectTransformXML(in interface{}, modifier pathModifier) (interface{}, error) {
+	path := ot.jsonPath
+	if modifier != nil {
+		path = modifier(path)
+	}
+	var newValue map[string]interface{}
+
+	// For the object use a transform if it exists or the default or an empty map
+	if ot.transforms != nil {
+		rawValue, err := ot.transforms.transform(in, "object", modifier, ot.format)
+		if err != nil {
+			return nil, err
+		}
+
+		switch v := rawValue.(type) {
+		case *xmlquery.Node:
```

Comment:
In the array transformer it works not by redefining `in` but by making a path modifier that is passed to all the children. The path modifier then only modifies relative paths. If I'm remembering the value of in correctly doesn't changing it out at this point break the possibility non-relative paths on transforms within the object? To state it another way if one of the fields in the object had a transform that should pull from another path then that path won't be found right?

If that is right then I think we need to first ask if that use case is needed and if it isn't document that anytime you specify a transform on an object all child transforms must be relative.

Reply:
i have documented the limitations and will make a ticket to revisit possibly resolving this in the future. As it stands now if an xmlPath is declared on an object all paths inside that object must be relative to the initial declared path, so things above the level of the declared path of the object will be inaccessible inside the object. I believe all SDI use cases will be covered as is.